src/Converter.cpp

```diff
@@ -141,21 +152,59 @@ int convertStreams(
 int convertSamples(Converter& conv, pocolog_cpp::InputDataStream* stream,
     const int start, const int end, const int verbose)
 {
+    
+    float next_report_progress = 0.0;
+    float report_progress_delta = 0.1;
+    
     for(size_t t = start; t < end; t++)
     {
-        std::vector<uint8_t> data;
-        const bool ok = stream->getSampleData(data, t);
+
+        if(verbose >= 2) {
+            std::cout << "[pocolog2msgpack] Converting sample #" << t
+                << std::endl;  
+        }
+        std::vector<uint8_t> curSampleData;
+        const bool ok = stream->getSampleData(curSampleData, t);
+        
         if(!ok)
         {
             std::cerr << "[pocolog2msgpack] ERROR: Could not read sample data."
                 << std::endl;
             return EXIT_FAILURE;
         }
-
-        if(verbose >= 2)
-            std::cout << "[pocolog2msgpack] Converting sample #" << t
-                << std::endl;
-        conv.convertSample(&data[0]);
+        
+        if(verbose >= 3) {        
+            std::cout << "Converting sample of size " << curSampleData.size() << std::endl;
+        }
+        
+        if(verbose >= 4){
+            for ( size_t i=0; i < curSampleData.size() ; i++ ){
+                if (i % 32 == 0) std::cout << std::endl;
+                std::ios_base::fmtflags f( std::cout.flags() );
+                std::cout << std::setfill('0') << std::setw(2) << std::right << std::hex << (unsigned int)(curSampleData[i]) << " ";
+                std::cout.flags( f );
+                std::cout << " ";
+                
+                if (i > 128) {
+                    std::cout << "...";
+                    break;
+                }
+            }
+            std::cout << std::endl;   
+        } 
+
+        conv.convertSample(curSampleData);
+        
+        float progress = ((float)(t-start))/(end-start);
+        
+        if ( progress >= next_report_progress ) {            
+            next_report_progress += report_progress_delta;
+            if (verbose >= 1) {
+                std::cout << "[pocolog2msgpack] " << (int)(progress*100) << "% of stream done." << std::endl;  
```

Comment:
A new feature. Should we document this somewhere?

Reply:
I would not call it a feature...but maybe adapt the output or verbosity level if you prefer (the stream names and number of samples are also printed by default).